### ISSUE TITLE
Inconsistent incorrect warnings with Style/TrailingCommaInLiteral warnings

### ISSUE THREAD

**Issue by DocX** [rubocop-hq/rubocop#5529]
Config:
```yaml
Style/TrailingCommaInArguments:
  EnforcedStyleForMultiline: "comma"
```

Having comma at the end of array fails (incorrectly) in this case:

```ruby
[
  {
    # ...
  }, {
    # ...
  },
]
```
```
C: Style/TrailingCommaInLiteral: Avoid comma after the last item of an array, unless each item is on its own line.
  },
   ^
```

... but is fine (incorrectly) with no comma at end:
```ruby
[
  {
    # ...
  }, {
    # ...
  }
]
```
```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

So OK - I will not add the trailing comma.

But then... I remove the second item, and it fails (correctly):

```ruby
[
  {
    # ...
  }
]
```
```
C: Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline array.
  }
```

also fails (correctly) in this case:
```ruby
[
  {
    # ...
  },
  {
    # ...
  }
]
```
```
C: Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline array.
  }
```

Versions:
Tested on `rubocop (0.50.0)` and `rubocop (0.52.1)`

**Comment by Geesu** [comment]
Ever solve this? Having the same problem...

**Comment by mikegee** [comment]
I'm confused because the initial report shows config for `Style/TrailingCommaInArguments` then offenses from `Style/TrailingCommaInLiteral`.

**Comment by DocX** [comment]
Hi @mikegee I must have made some mistake in copy pasting it last time. Here again better:

`rubocop_test.rb`:
```rb
# frozen_string_literal: true

def multi_item_array
  [
    {
      a: 12,
    }, {
      b: 33,
    }, {
      c: 11,
    },
  ]
end

def single_item_array
  [
    {
      a: 12,
    }
  ]
end
```

Result:
```
$ rubocop app/commands/educate/article/rubocop_test.rb
Inspecting 1 file
C

Offenses:

app/commands/educate/article/rubocop_test.rb:11:6: C: Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array, unless each item is on its own line.
    },
     ^
app/commands/educate/article/rubocop_test.rb:19:5: C: Style/TrailingCommaInArrayLiteral: Put a comma after the last item of a multiline array.
    }
    ^

1 file inspected, 2 offenses detected
```

**Comment by DocX** [comment]
Hm, or is the idea that we should use this syntax - with each hash starting on its own line?
```rb
def multi_item_array2
  [
    {
      a: 12,
    },
    {
      b: 33,
    },
    {
      c: 11,
    },
  ]
end
```

**Comment by ryanhageman** [comment]
Is this still an issue?

**Comment by mxmlnglt** [comment]
I can confirm this is still an issue, especially with single item array.
